src/sysctl/sysctl.c

```diff
@@ -26,25 +26,71 @@ static char **arg_prefixes = NULL;
 static bool arg_cat_config = false;
 static bool arg_no_pager = false;
 
+typedef struct Option {
+        char *key;
+        char *value;
+        bool ignore_failure;
+} Option;
+
+static Option *option_free(Option *o) {
+        if (!o)
+                return NULL;
+
+        free(o->key);
+        free(o->value);
+
+        return mfree(o);
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(Option*, option_free);
+DEFINE_HASH_OPS_WITH_VALUE_DESTRUCTOR(option_hash_ops, char, string_hash_func, string_compare_func, Option, option_free);
```

Comment:
This macro is defined in a following commit. That means that a build of this commit--e.g., during bisection--will fail. Please reorder the commits to avoid that.

Reply:
Sure, done.